### PR TITLE
Add fuzzy-finder for project switching

### PR DESCRIPTION
**Pull request by ChrisPenner**
switch-fzf

## Overview

Adds fuzzy-finder for `switch` without any arguments which lists all project/branch pairs for you to search, whatever you select will be switched to.

## Implementation notes

* Make the project-branch optional for switch, if it's `Nothing`, trigger a fuzzy-find to fill it in.
* Adds a new query to fetch ALL project/branch pairs. `fzf` is killer fast, so should have no problem handling thousands of branches if it comes to that.

## Interesting/controversial decisions

Should it search projects, then search branches instead?

## Test coverage

None

### CODE REVIEW

**Comment by ceedubs** [comment]
@ChrisPenner nice! I will definitely be using this.

Would it be much effort and/or desirable to put the current project at the beginning (or end? Which is it?) of the list? My thought was that I'm often switching between `/main` and a `/my-feature-branch` and I might just type `switch` and then hit the up button. Maybe bonus points for taking the recency of visiting a branch into account, but that's getting fancy 😄.

File: unison-cli/src/Unison/Codebase/Editor/Input.hs [hunk 1]
```diff
@@ -230,7 +230,7 @@ data Input
   | DiffNamespaceToPatchI DiffNamespaceToPatchInput
   | ProjectCreateI Bool {- try downloading base? -} (Maybe ProjectName)
   | ProjectRenameI ProjectName
-  | ProjectSwitchI ProjectAndBranchNames
+  | ProjectSwitchI (Maybe ProjectAndBranchNames {- Nothing triggers fuzzy-finder -})
```
Comment (aryairani) [inline unison-cli/src/Unison/Codebase/Editor/Input.hs:233]:
```suggestion
  | ProjectSwitchI (Maybe ProjectAndBranchNames {- `Nothing` triggers fuzzy-finder -})
```

File: unison-cli/src/Unison/CommandLine/InputPatterns.hs [hunk 2]
```diff
@@ -686,20 +686,26 @@ deleteGen :: Maybe String -> String -> ([Path.HQSplit'] -> DeleteTarget) -> Inpu
...
+            ( P.sep
+                " "
+                [ backtick (P.sep " " [P.string cmd, "foo bar"]),
+                  "removes the",
+                  P.string target,
+                  "name `foo` and `bar` from the namespace."
+                ],
+              ""
+            )
+          ]
```
Comment (aryairani) [inline unison-cli/src/Unison/CommandLine/InputPatterns.hs:708]: Delayed ormolu?

**Review by aryairani — APPROVED** [review]

**Comment by aryairani** [comment]
Has a merge conflict now but otherwise looks good 😬

**Comment by ChrisPenner** [comment]
@ceedubs Okay I put the active project at the bottom :) I'll leave sorting by "frecency" for another day

**Pull request merged**